### PR TITLE
Remove static abort from AbortSignal

### PR DESCRIPTION
**Pull request by sandersn**
@types/node doesn't have it, and it's not worth it to make @types/node create a new typingsVersion.

### CODE REVIEW

File: inputfiles/removedTypes.jsonc [hunk 1]
```diff
@@ -166,6 +166,14 @@
             "Text": {
                 "implements": ["GeometryUtils"]
             },
+            // This is hard to get in sync with @types/node
+            "AbortSignal": {
+                "methods": {
+                    "method": {
+                        "abort": null
```
Comment (saschanaz) [inline inputfiles/removedTypes.jsonc:173]: I think this one should be preferred over node, as Node is just following the web standard? What happened there?

**Comment by SimonSchick** [comment]
But node does have this method?
```
Type ".help" for more information.
> AbortSignal.abort()
AbortSignal { aborted: true }
```

**Comment by saschanaz** [comment]
> But node does have this method?

Web browsers and Node.js have the method defined as https://dom.spec.whatwg.org/#interface-AbortSignal .

**Comment by orta** [comment]
Yeah, agree - its more that we didn't think adding this was worth forcing `@types/node` to have to switch to TS 4.5 for their types and then requiring _many_ projects to update for this one property. Not that it's incorrect

**Comment by saschanaz** [comment]
Can it be fixed locally in TS rather than affecting @types/web?

**Comment by orta** [comment]
It's a normal assignability bug, not a change in TypeScript (you can see a [tonne in here](https://github.com/typescript-bot/TypeScript/pull/197/files#diff-01b0ab09324fb295447d930f7e8edddbe4093fa413b59a6d2bfc2d66762dc712R3))

```
node_modules/@types/node/globals.d.ts(85,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'AbortSignal' must be of type '{ new (): AbortSignal; prototype: AbortSignal; abort(): AbortSignal; }', but here has type '{ new (): AbortSignal; prototype: AbortSignal; }'.
```

Basically it means that anyone with an older version of `@types/node` cannot upgrade to TS 4.5 without updating their node types, and it means that the `@types/node` would need to bump their minimum TS version to 4.5 for this one property

**Comment by saschanaz** [comment]
I mean, we'll have to add that to node one day, so it would be good if TS have a local patch that removes abort() so that anyone need it can install @types/web?

**Comment by saschanaz** [comment]
That can happen in deploy script IMO.

**Comment by SimonSchick** [comment]
Can we not change the type of `AbortSignal` to be an interface and use declaration merging?

**Comment by saschanaz** [comment]
> Can we not change the type of `AbortSignal` to be an interface and use declaration merging?

Sounds like an option tbh.

**Comment by sandersn** [comment]
Does the code have the ability to generate an interface for the static side too? If not, I guess we could add that.

**Comment by saschanaz** [comment]
No such ability, I think AbortSignal can be special cased.

**Comment by sandersn** [comment]
oh, I forgot about addedTypes.jsonc. That should work.

**Comment by orta** [comment]
So, I think we're going to have to take this PR as-is because I don't think we're going to get a good answer which doesn't force users to update node or force a typesVersion on node before the release today. We shipped a build to TypeScript _with_ these changes in, because we don't want to force the node update on anyone migrating to the new TS build, and I want to have corresponding `@types/web` to refer to it.

I'll make a PR reverting it after we're sure the RC dom is going to be what we ship in TS. 

I brought up https://github.com/microsoft/TypeScript/issues/43972 in a TS language design meeting on Wednesday and there was general agreement to explore finding the common subset and extract it in here which is going to probably be tricky work in this repo. https://github.com/microsoft/TypeScript/issues/45328

**Comment by saschanaz** [comment]
> I want to have corresponding `@types/web` to refer to it.

I'd prefer manual patching in deploy script if this isn't a strict requirement.

**Comment by orta** [comment]
It's not

**Comment by saschanaz** [comment]
Looks good, please reformat 😉

**Comment by saschanaz** [comment]
LGTM

File: deploy/createTypesPackages.js [hunk 2]
```diff
@@ -188,6 +189,30 @@ function prependAutoImports(pkg, packagePath) {
...
+/**
+ * Handles any post-processing we do for deployment.
+ * @param {Package} pkg
+ * @param {URL} packagePath
+ */
+function postProcessDTSFiles(pkg, packagePath) {
+  iterateThroughFiles((content) => {
+    return content.replace(
+      "abort(): AbortSignal;",
+      "// abort(): AbortSignal; - To be re-added in the future"
```
Comment (saschanaz) [inline deploy/createTypesPackages.js:201]: This maaayy advertise `@types/web` but your call really.

**Comment by orta** [comment]
OK, let's get this in - which will unblock merging some of the other PRs

**Pull request merged**